src/sentry/options/manager.py

```diff
@@ -81,53 +105,78 @@ def set(self, key, value):
         )
 
         try:
-            self.update_cached_value(key, value)
+            self.update_cached_value(opt.cache_key, value)
+            return True
         except Exception:
             self.logger.warn(CACHE_UPDATE_ERR, key, exc_info=True)
+            return False
+
+    def lookup_key(self, key):
+        try:
+            return self.registry[key]
+        except KeyError:
+            # HACK: Historically, Options were used for random adhoc things.
+            # Fortunately, they all share the same prefix, 'sentry:', so
+            # we special case them here and construct a faux key until we migrate.
+            if key[:7] == 'sentry:':
+                self.logger.info('Using legacy key: %s', key, exc_info=True)
+                # History shows, there was an expectation of no types, and empty string
+                # as the default response value
+                return Key(key, '', object, self.FLAG_DEFAULT, self._make_cache_key(key))
+            raise UnknownOption(key)
 
     def get(self, key):
```

Comment:
It would probably be a good idea to perform type validation on the result of this method as well — the value may been have been entered in a configuration file incorrectly, or serialized wrong, etc. — having a safety net here could prevent some headaches later.